soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -260,328 +270,189 @@ CrowdControl::Effect* CrowdControl::ParseMessage(char payload[512]) {
     effect->lastExecutionResult = EffectResult::Initiate;
     effect->id = dataReceived["id"];
     auto parameters = dataReceived["parameters"];
+    auto effectName = dataReceived["code"].get<std::string>();
+
     if (parameters.size() > 0) {
-        effect->value = dataReceived["parameters"][0];
+        effect->value[0] = dataReceived["parameters"][0];
```

Comment:
noticed this earlier, it's not clear what `value[0]` is here, it might be nice to use a little enum instead of just `1` and `0` so it's easier to understand what's happening

Reply:
This is a little difficult because what this means differs from effect to effect. For example, *most* CC effects that set a value are just the amount of whatever the effect does. The amount of hearts healed, how strong is the knockback, etc. In the case of enemy spawning, this dictates the enemy ID that needs to be spawned, where value[1] is used for the enemy parameter instead.

I'm open for ideas on this one.

Reply:
hmm, i'm also not sure then, it'd be nice figure out a way for it to be a little more clear but i guess we can worry about this part later